src/content/docs/en/guides/cms/keystatic.mdx

```diff
@@ -2,13 +2,263 @@
 title: Keystatic & Astro
 description: Add content to your Astro project using Keystatic as a CMS
 type: cms
-stub: true
 service: Keystatic
 i18nReady: false
 ---
+import FileTree from '~/components/FileTree.astro'
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import Aside from '~/components/Aside.astro'
 
 [Keystatic](https://keystatic.com/) is an open source, headless content-management system that allows you to structure your content and sync it with GitHub.
 
+:::tip
+If you're starting a **new Astro + Keystatic project from scratch**, you can use the [Keystatic CLI](https://keystatic.com/docs/quick-start#keystatic-cli) to generate a new project in seconds:
+
+```bash
+npm create @keystatic@latest
+```
+
+Select the Astro template, and you'll be ready to [deploy](#deploying-keystatic--astro)!
+:::
+
+## Prerequisites
+
+This guide assumes you already have **an existing Astro project** in SSR running on [`output: 'hybrid'`](<>). If you don’t, you can follow the instructions in our [Installation guide](/en/install/auto/).
+
+:::note
+If you intend to sync Keystatic's data with GitHub, you will also need **a GitHub account with `write` permissions** on the repository for this project.
+:::
+
+## Installing dependencies
+
+You're going to need to install a few packages to get Keystatic going.
+
+Keystatic outputs Markdoc and has a dependency on React, so first you need to add Astro's first-party integrations for both:
+
+```bash
+npx astro add react markdoc
+```
+
+You will also need two Keystatic packages:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install @keystatic/core @keystatic/astro
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install @keystatic/core @keystatic/astro
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add @keystatic/core @keystatic/astro
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+## Creating a Keystatic config file
+
+Keystatic needs a config file. This is where you can connect a project with a specific GitHub repository and define a content schema.
```

Comment:
```suggestion
Keystatic needs a config file. This is where you can connect a project with a specific GitHub repository and/or define a content schema.
```

Not sure if I like the `and/or` syntax, but technically you dont' *need* to connect the project to GitHub. You need however to define some content schemas before yous can do anything with Keystatic.

Reply:
OK, then what about something like (include an extra bit at the end if you like something like that, or just stop early):

> A Keystatic config file is required to define your content schema. This file will also allow you to connect a project to a specific GitHub repository (if you decide to do so).

And fwiw, technically in the original version, we're only saying above that the CONFIG file is required in order to do these things, not that these things are both required things to do! But, separation like this can help you avoid an and/or situation.)

Reply:
Yes this is great!

```suggestion
A Keystatic config file is required to define your content schema. This file will also allow you to connect a project to a specific GitHub repository (if you decide to do so).
```